andela/codepirates-ah-frontend
# #167313379 Highlight share and comment highlights

#### What does this PR do?

Add highlights feature

#### Description of Task to be completed?

- An authenticated user should be able to create highlights
- If unauthenticated and attempts to save highlight, should be redirected to login and back to the article of interest after login
 
#### How should this be manually tested?

- Login and make any selection in the article body
- In the dialogue that pops up hit the highlight button
- Observer the highlight
- Log out and observe that the highlights are not showed to the unauthenticated or authenticated as otherwise
- Log back into your account and observe that your highlights were persisted to the database
- Change the browser or device to confirm highlights are persisted to remote storage (database) 

#### Any background context you want to provide?

- Supposed to implement sharing and commenting in relation to highlights as well
- Will add these in the ampleness of time

#### What are the relevant pivotal tracker stories?

[finishes #16731337](https://www.pivotaltracker.com/story/show/167313379)

#### Screenshots (if appropriate)

![image](https://user-images.githubusercontent.com/36167006/68850557-ea32f000-06dc-11ea-8b55-37cbf63a7178.png)


#### Questions:
